clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -19,6 +19,15 @@
  * If a valid partition number is specified that partition will be used when sending the record. If no partition is
  * specified but a key is present a partition will be chosen using a hash of the key. If neither key nor partition is
  * present a partition will be assigned in a round-robin fashion.
+ * <p>
+ * The record also has an associated timestamp. If user did not provide a timestamp, the producer will stamp the record
+ * with a timestamp depending on what is the timestamp type used by the topic.
+ * <li>
+ * If the topic is configured to use {@link org.apache.kafka.common.record.TimestampType#CreateTime CreateTime}
+ * the timestamp will be user specified timestamp or the producer's current time.
+ * <li>
+ * If the topic is configured to use {@link org.apache.kafka.common.record.TimestampType#LogAppendTime LogAppendTime}
```

Comment:
Are these comments correct? The timestamp in ProducerRecord is always set by the producer.